### PR TITLE
Fix URL to URI

### PR DESCRIPTION
**Pull request by shucon**
Partial-PR: https://github.com/SeleniumHQ/selenium/issues/8020

**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by shucon** [comment]
@shs96c @AutomatedTester

File: java/server/src/org/openqa/selenium/grid/graphql/selenium-grid-schema.graphqls [hunk 1]
```diff
@@ -23,7 +24,9 @@ type GridQuery {
 }
 
 type Grid {
-  url: Url!
+  uri: Uri!
+  host: String!
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/selenium-grid-schema.graphqls:28]: The URI contains both the host and the port. Please don't duplicate the information

File: java/server/src/org/openqa/selenium/grid/graphql/Grid.java [hunk 2]
```diff
@@ -23,23 +23,32 @@
...
     this.distributorStatus = Suppliers.memoize(distributor::getStatus);
   }
 
-  public String getUrl() {
-    return url;
+  public URI getUri() {
+    return uri;
+  }
+
+  public String getHost() {
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/Grid.java:46]: We don't need to expose this. Please don't duplicate the information.

File: java/server/src/org/openqa/selenium/grid/graphql/Grid.java [hunk 3]
```diff
@@ -23,23 +23,32 @@
...
+    return uri.getHost();
+  }
+
+  public int getPort() {
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/Grid.java:50]: As above.

File: java/server/src/org/openqa/selenium/grid/graphql/Node.java [hunk 4]
```diff
@@ -26,14 +26,18 @@
 
   private final UUID id;
   private final URI uri;
+  private final String host;
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/Node.java:29]: We don't need to expose this.

File: java/server/src/org/openqa/selenium/grid/graphql/Node.java [hunk 5]
```diff
@@ -26,14 +26,18 @@
...
   private final boolean isUp;
   private final int maxSession;
+  private final int port;
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/Node.java:32]: Or this.

File: java/server/src/org/openqa/selenium/grid/graphql/Node.java [hunk 6]
```diff
@@ -26,14 +26,18 @@
...
 
   public Node(UUID id, URI uri, boolean isUp , int maxSession) {
     this.id = Require.nonNull("Node id", id);
     this.uri = Require.nonNull("Node uri", uri);
     this.isUp = isUp;
     this.maxSession = Require.nonNull("Node maxSession", maxSession);
+    this.host = uri.getHost();
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/Node.java:39]: Don't keep the values of derived fields as fields themselves unless calculating them is expensive. And we don't need this in this case.

File: java/server/src/org/openqa/selenium/grid/graphql/selenium-grid-schema.graphqls [hunk 7]
```diff
@@ -3,7 +3,6 @@ schema {
 }
 
 scalar Uri
-scalar Url
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/selenium-grid-schema.graphqls:6]: 🎉

File: java/server/src/org/openqa/selenium/grid/graphql/selenium-grid-schema.graphqls [hunk 8]
```diff
@@ -14,6 +13,8 @@ enum Status {
 type Node {
   id: ID!
   uri: Uri!
+  host: String!
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/graphql/selenium-grid-schema.graphqls:16]: This is duplicating information from the `uri` field. Please remove `host` and `port`

**Comment by shs96c** [comment]
This looks great now. Thank you for the changes, and thank you for pushing this forwards :)

**Pull request merged**